controllers/v1/chat/index.js

```diff
@@ -0,0 +1,116 @@
+const { CREATED, UNAUTHORIZED, BAD_REQUEST, OK } = require("http-status-codes");
+
+//DB
+const Chat = require('../../../model/v1/Chat');
+
+//Validation
+const validateChat = require("../../../validators/chat");
+
+const {
+  successWithData,
+  successNoData,
+} = require("../../../utils/successHandler");
+
+// Error
+const AppError = require("../../../utils/appError");
+
+exports.createChat = async (req, res, next) => {
+  try {
+    const users1 = [req.user.id, req.body.receiverId];
+    const users2 = [req.body.receiverId, req.user.id];
+    
+    const {errors, isValid } = validateChat(users1);
+
+    if (req.user.id == req.body.receiverId) {
+      errors.msg = "A user cannot create a chat with themself";
+      return AppError.validationError(res, UNAUTHORIZED, errors);
+    }
+
+    const chatExists = await Chat.findOne({ $or: [ {users:users1}, {users:users2} ] });
+    if (chatExists) {
+      errors.msg = "A chat between these users already exists";
+      return AppError.validationError(res, UNAUTHORIZED, errors);
+    }
+    const newChat = await Chat.create({users:users1});
+    return successWithData(
+      res,
+      CREATED,
+      "Chat created successfully",
+      newChat
+    );
+  } catch (err) {
+    console.log(err);
+    return AppError.tryCatchError(res, err);
+  }
+};
+
+exports.acceptChat = async (req, res, next) => {
```

Comment:
call the name of this controller to something else, we can use this controller to block different chatrooms, by turning the `accepted` field back to false!!